frontend/app/handlers.ts

```diff
@@ -1,19 +1,47 @@
+import {
+    crypto_pwhash_str,
+    crypto_pwhash_str_verify,
+    crypto_pwhash_OPSLIMIT_INTERACTIVE,
+    crypto_pwhash_MEMLIMIT_INTERACTIVE,
+} from "libsodium-wrappers-sumo";
+
 // Make API call to server to check if password was found in breached dataset
 export const checkSecurity = async (password: string) => {
     try {
-        const response = await fetch("http://localhost:18080/intersection", {
-            method: "POST",
-            mode: "cors",
-            headers: {
-                "Access-Control-Allow-Headers": "*", // cors setting
-                "Content-Type": "application/json"
-            },
-            body: password
-        })
+        // hash the password
+        const passwordHash = crypto_pwhash_str(
+            password,
+            crypto_pwhash_OPSLIMIT_INTERACTIVE,
+            crypto_pwhash_MEMLIMIT_INTERACTIVE
+        );
+        if (!crypto_pwhash_str_verify(passwordHash, password)) {
```

Comment:
**question**: should we include the checking if out of memory (as shown in the code below)? Or is that covered by the `crypto_pwhash_str_verify(passwordHash, password)` check?

`if (crypto_pwhash_str
    (hashed_password, PASSWORD, strlen(PASSWORD),
     crypto_pwhash_OPSLIMIT_SENSITIVE, crypto_pwhash_MEMLIMIT_SENSITIVE) != 0) {
    /* out of memory */
}`